internal/service/securityhub/configuration_policy.go

```diff
@@ -0,0 +1,620 @@
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).SecurityHubClient(ctx)
+
+	input := &securityhub.GetConfigurationPolicyInput{
+		Identifier: aws.String(d.Id()),
+	}
+
+	out, err := conn.GetConfigurationPolicy(ctx, input)
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] Security Hub Configuration Policy (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return diags
+	}
+
+	if err != nil {
+		return sdkdiag.AppendErrorf(diags, "reading Security Hub Configuration Policy (%s): %s", d.Id(), err)
+	}
+
+	d.Set("name", out.Name)
+	d.Set("description", out.Description)
+	d.Set("arn", out.Arn)
+	if err := d.Set("policy_member", []interface{}{flattenSecurityHubPolicy(out.ConfigurationPolicy)}); err != nil {
+		return sdkdiag.AppendErrorf(diags, "setting policy_member: %s", err)
+	}
+
+	return diags
+}
+
+func resourceConfigurationPolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).SecurityHubClient(ctx)
+
+	input := &securityhub.DeleteConfigurationPolicyInput{
+		Identifier: aws.String(d.Id()),
+	}
+
+	_, err := conn.DeleteConfigurationPolicy(ctx, input)
+	if err != nil {
+		return sdkdiag.AppendErrorf(diags, "deleting Security Hub Configuration Policy (%s): %s", d.Id(), err)
+	}
+
+	return diags
+}
+
+// validatePolicyMember performs validation before running creates/updates to prevent certain issues with state.
+func validatePolicyMember(apiPolicy *types.PolicyMemberSecurityHub) error {
+	// security_controls_configuration can be specified in Creates/Updates and accepted by the APIs,
+	// but the resources returned by subsequent Get API call will be nil instead of non-nil.
+	// This leaves terraform in perpetual drift and so we prevent this case explicitly.
+	if !*apiPolicy.Value.ServiceEnabled && apiPolicy.Value.SecurityControlsConfiguration != nil {
+		return errors.New("security_controls_configuration cannot be defined when service_enabled is false")
+	} else if *apiPolicy.Value.ServiceEnabled && apiPolicy.Value.SecurityControlsConfiguration == nil {
+		return errors.New("security_controls_configuration must be defined when service_enabled is true")
+	}
+
+	// If ServiceEnabled is true, then Create/Update APIs require exactly one of enabled or disable control fields to be non-nil.
+	// If terraform defaults are set for both, then we choose to set DisabledSecurityControlIdentifiers to the empty struct.
+	if *apiPolicy.Value.ServiceEnabled && apiPolicy.Value.SecurityControlsConfiguration != nil &&
+		apiPolicy.Value.SecurityControlsConfiguration.DisabledSecurityControlIdentifiers == nil &&
+		apiPolicy.Value.SecurityControlsConfiguration.EnabledSecurityControlIdentifiers == nil {
+		apiPolicy.Value.SecurityControlsConfiguration.DisabledSecurityControlIdentifiers = []string{}
+	}
+
+	return nil
+}
+
+func expandSecurityHubPolicy(tfMap map[string]interface{}) *types.PolicyMemberSecurityHub {
```

Comment:
whoops. Tried to re-name SecurityHubPolicy to PolicyMember to avoid using name of service. Will want to update these `expand` and `flatten` methods too if decide to go through with that re-name.

Reply:
We prefer using the underlying AWS type in the `expand` and `flatten` names so I'll rename to `expandPolicyMemberSecurityHub`.